src/pages/projectView/projectMainBody/sections/Rewards.tsx

```diff
@@ -45,7 +47,16 @@ export const Rewards = forwardRef<HTMLDivElement>((_, ref) => {
               reward={reward}
               onRewardClick={() => {
                 const count = (fundFormState.rewardsByIDAndCount && fundFormState.rewardsByIDAndCount[`${reward.id}`]) || 0
-                updateReward({ id: toInt(reward.id), count: count + 1 })
+                const rewardStockRemaining = reward.maxClaimable ? reward.maxClaimable - reward.sold : 100;
+                if(rewardStockRemaining > count) {
+                  updateReward({ id: toInt(reward.id), count: count + 1 })
+                } else {
+                  toast({
+                    title: 'Reward Limit',
+                    description: `Maximum number of ${rewardStockRemaining} available for this reward`,
+                    status: 'error',
+                  })
+                }
```

Comment:
This part is repeated here again, maybe there is a way to standardize this, like a validateRewardAvailability function.

This is business logic basically, and having the same thing in multiple places creates overhead later to change all of it when the logic is updated.